### ISSUE TITLE
HttpTrigger sauthentication not orking

### ISSUE THREAD

**Issue by SteveALee** [Azure/azure-functions-cli#29]

**Comment by ahmelsayed** [comment]
At the moment all [auth is disabled when running locally](https://github.com/Azure/azure-webjobs-sdk-script/blob/split-cli/src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs#L47-L49), so it is intentional.
The challenge locally is that you won't have UI for managing keys for the function\host and I don't have any commands for managing that either. When I thought about it a while ago, I couldn't think of scenarios where you'd need auth when working locally. But I'm open to suggestions

**Comment by SteveALee** [comment]
Worth adding to readme?

> I couldn't think of scenarios where you'd need auth when working locally. But I'm open to suggestions

I generally agree, especially as and key will be ignored.

I can only think of one case: when including a test to check that an incorrect api key fails. I guess it's a mute point if this should be in unit tests or integrations tests (probably some sort of module test), but as it is such a test will fail locally but pass remotely. We'd need to reserve these for tests against the live server. As you allow local testing it seems a shame not to be able to use the same test suite for the entire CD chain.

I'm probably overthinking it. Especially as the only way to get a function ID is manually from the portal, AFAIK .

**Comment by lindydonna** [comment]
Yeah, I don't see a big need for this scenario. Changing bug title to reflect the ask.

**Comment by rasmuschristensen** [comment]
This would make great sense when AuthLevel.User is working. Then you might need some info from claims based on the token provided. 
For other cases it's just to say okay Auth is working before you deploy it online

**Comment by mvanbeusekom** [comment]
@lindydonna, @SteveALee, so now it is also possible to add Azure Functions as a Docker container the need to be able to get access to function keys (and possibly also the master and host keys) becomes a lot bigger. 

In my experience when I host the Azure Function in a Docker container the authentication works. Meaning when I configure my function to require a function key and run the container locally, I get a 401 Unauthorized response. Since there is no way (at least to my knowledge, see also my Stack Overflow question) to get local access to the keys, I don't know which key to supply to get it working. Only until I publish my container to Azure I will be able to get access to keys. Meaning there is no way of testing my functionality (unless I keep on changing permissions) when running in a Docker container.

This also means these type of authorization will not work when hosting everything on premise (without the use of Azure Stack) or on other cloud providers (which I can imagine is not something Microsoft is really worried about).

In short, I would love to have this feature available and think there are more scenarios then only local testing.

**Comment by VenkateshSrini** [comment]
For the same of on-premise hosting I would like to have at least adding such keys via JSON file and deploy the same in my container. Is that possible at least

**Comment by adrianrbezzina** [comment]
Is there any progress on how to consume HttpTriggers with authentication codes from within containers? Documentation relating to hosting Azure Functions in Containers (via Kubernetes) are available, though these neglects to mention how to consume HttpTriggers with authentication codes: https://github.com/Azure/azure-functions-core-tools#getting-started-on-kubernetes

**Comment by SpicySyntax** [comment]
> Is there any progress on how to consume HttpTriggers with authentication codes from within containers? Documentation relating to hosting Azure Functions in Containers (via Kubernetes) are available, though these neglects to mention how to consume HttpTriggers with authentication codes: https://github.com/Azure/azure-functions-core-tools#getting-started-on-kubernetes

Has this progressed any futher? I am not sure how to access or set the keys for a function I am deploying to a Azure Kubernetes Service Cluster

**Comment by VenkateshSrini** [comment]
Hey,

Please see this thread it will help you. I have tested it locally and it works. I'm yet to test the same on container
https://github.com/Azure/azure-functions-host/issues/4147#issuecomment-477442831

**Comment by jyotirmay123** [comment]
> @lindydonna, @SteveALee, so now it is also possible to add Azure Functions as a Docker container the need to be able to get access to function keys (and possibly also the master and host keys) becomes a lot bigger.
> 
> In my experience when I host the Azure Function in a Docker container the authentication works. Meaning when I configure my function to require a function key and run the container locally, I get a 401 Unauthorized response. Since there is no way (at least to my knowledge, see also my [Stack Overflow](https://stackoverflow.com/questions/53126627/http-trigger-azure-function-in-docker-with-non-anonymous-authlevel) question) to get local access to the keys, I don't know which key to supply to get it working. Only until I publish my container to Azure I will be able to get access to keys. Meaning there is no way of testing my functionality (unless I keep on changing permissions) when running in a Docker container.
> 
> This also means these type of authorization will not work when hosting everything on premise (without the use of Azure Stack) or on other cloud providers (which I can imagine is not something Microsoft is really worried about).
> 
> In short, I would love to have this feature available and think there are more scenarios then only local testing.

Still not available..  Such a shame.

**Comment by asos-alexhaigh** [comment]
> > Is there any progress on how to consume HttpTriggers with authentication codes from within containers? Documentation relating to hosting Azure Functions in Containers (via Kubernetes) are available, though these neglects to mention how to consume HttpTriggers with authentication codes: https://github.com/Azure/azure-functions-core-tools#getting-started-on-kubernetes
> 
> Has this progressed any futher? I am not sure how to access or set the keys for a function I am deploying to a Azure Kubernetes Service Cluster


I found [this PR](https://github.com/Azure/azure-functions-host/pull/4462?source=post_page-----b12df2933be4---------------------------------------) which helped me figure out how to set the keys on Kubernetes.

**Comment by satvu** [comment]
This has been addressed by the `--enableAuth` flag on `func start`, which enforces function key authentication locally (a bug where it was bypassed was fixed in PR #4211, Jan 2025).

For documentation on usage, see #3394. For improving the local key retrieval experience (func keys list), see #5312.

Closing as resolved.
